query/math.go

```diff
@@ -30,6 +30,16 @@ type mathTree struct {
 	Child []*mathTree
 }
 
+var (
+	ErrorIntOverflow     = errors.New("Integer overflow")
+	ErrorFloatOverflow   = errors.New("Float overflow")
```

Comment:
Floats don't overflow. Since IEEE 754 is already best-effort and widely used, I think it should be fine to remove all floating-point overflow checks.